CLI/src/CLI.cs

```diff
@@ -36,18 +36,14 @@ class CLI
                 debugLine += Path.GetFileName(config.InputFiles[0]);
             }
             debugLine += "\n";
-            File.AppendAllText("TypeCobol.CLI.log", debugLine);
+            //Use user-defined log path if -log option used, otherwise use default location for log file
+            File.AppendAllText(config.LogFile ?? TypeCobolConfiguration.LogFileName, debugLine);
             Console.WriteLine(debugLine);
-
-            TextWriter textWriter;
-            if (config.ErrorFile == null) textWriter = Console.Error;
-            else textWriter = File.CreateText(config.ErrorFile);
-            AbstractErrorWriter errorWriter;
-            if (config.IsErrorXML) errorWriter = new XMLWriter(textWriter);
-            else errorWriter = new ConsoleWriter(textWriter);
+            TextWriter textWriter = config.ErrorFile == null ?  Console.Error : File.CreateText(config.ErrorFile);
+            AbstractErrorWriter errorWriter = config.IsErrorXML ? (AbstractErrorWriter) new XMLWriter(textWriter) : new ConsoleWriter(textWriter);
```

Comment:
You can remove the cast to (AbstractErrorWriter)

Reply:
C# compiler doesn't allow for this ternary operation without cast to first return value, since XMLWriter and ConsoleWriter do not implicitly cast to the same object class.